sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -122,6 +122,14 @@ class SparkSession private(
   val sqlContext: SQLContext = new SQLContext(this)
 
   /**
+   * This is the interface through which the user can get [[JavaSparkContext]]
+   *
+   * @since 2.0.0
+   */
+  @transient
+  val javaSparkContext: JavaSparkContext = new JavaSparkContext(sparkContext)
```

Comment:
@srowen Yeah you are right but for java developers its pain when they get sparkContext from spark session and call parallelize method on it and its give error that its take argument type of any scala seq so i think for java developer or user its more easy to get api.